core/src/util/builders.cpp

```diff
@@ -288,12 +288,20 @@ bool isOutsideTile(const glm::vec3& _a, const glm::vec3& _b) {
     return false;
 }
 
-void buildPolyLineSegment(const Line& _line, PolyLineBuilder& _ctx) {
+void buildPolyLineSegment(const Line& _line, PolyLineBuilder& _ctx,
+                          size_t startIndex, size_t endIndex, bool endCap = true) {
 
-    int lineSize = (int)_line.size();
+    size_t origLineSize = _line.size();
+
+    // endIndex/startIndex could be wrapped values, calculate lineSize accordingly
+    int lineSize = (int)((endIndex > startIndex) ?
+                   (endIndex - startIndex) :
+                   (origLineSize - startIndex + endIndex));
```

Comment:
While loop can be used but still to calculate `v` value, tricky arithmetic needs to be done. 
I do not see much difference between using `while` vs `for` here.

P.S. BTW its not that tricky .. just a modulo operator :D